packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -0,0 +1,389 @@
+      location: vscode.ProgressLocation.Notification,
+      title: "Calculating packages to install or upgrade ...",
+      cancellable: false,
+    },
+    async () => {
+      const packageQuickPickItems = [];
+      if (action === InstallAction.Upgrade || action === InstallAction.Install) {
+        const minimumRequisites = selectedEmitter.requisites
+          ? ` Minimum requisites: install ${selectedEmitter.requisites?.join(", ")}`
+          : "";
+        const moreDetail = selectedEmitter.sourceRepo
+          ? `[**More Detail**](${selectedEmitter.sourceRepo})`
+          : "";
+        let packageFullName = selectedEmitter.package;
+        if (version) {
+          packageFullName = `${selectedEmitter.package}@${version}`;
+        }
+        packageQuickPickItems.push({
+          label: `${selectedEmitter.package}`,
+          detail: `TypeSpec library for emitting ${selectedEmitter.language} from TypeSpec files.${minimumRequisites} ${moreDetail}`,
+          packageFullName: packageFullName,
+          picked: true,
+        });
+        packagesToInstall.push(packageFullName);
+      }
+
+      for (const p of packagesToInstall) {
+        /* verify dependency packages. */
+        try {
+          const dependenciesToInstall = await npmUtil.calculateNpmPackageDependencyToUpgrade(
+            p,
+            version,
+            [npmDependencyType.dependencies, npmDependencyType.peerDependencies],
+          );
+          if (dependenciesToInstall.length > 0) {
+            packagesToInstall.push(...dependenciesToInstall.map((d) => `${d.name}@${d.version}`));
+            for (const dep of dependenciesToInstall) {
+              const packageFullName = `${dep.name}@${version ?? "latest"}`;
+              packageQuickPickItems.push({
+                label: `${dep.name}`,
+                detail: "Required for enabling the emitter library.",
+                packageFullName: packageFullName,
+                picked: true,
+              });
+            }
+          }
+        } catch (err) {
+          logger.error(`Exception occurred when check dependency packages for ${p}.`);
+        }
+      }
+      return packageQuickPickItems;
+    },
+  );
+
+  if (installPackageQuickPickItems.length > 0) {
+    const selectedPackages = await vscode.window.showQuickPick(installPackageQuickPickItems, {
+      title: "Install or update libraries",
+      canPickMany: true,
+      placeHolder: "Here are libraries to install or update.",
+      ignoreFocusOut: true,
+    });
+    if (!selectedPackages || selectedPackages.length === 0) {
+      logger.info("No package selected. Generating Cancelled.", [], {
+        showOutput: true,
+        showPopup: true,
+      });
+      return;
+    }
+    /* npm install packages. */
+    if (selectedPackages.length > 0) {
+      const installPackages = selectedPackages.map((p) => p.packageFullName);
+      logger.info(`Install ${installPackages.join("\n\n")} under directory ${baseDir}`, [], {
```

Comment:
no need to popup notification for this which is explicitly selected by user. And also there would already be another progress notification shown at the same time.